archspec/cli.py

```diff
@@ -22,3 +23,37 @@ def main():
 def cpu():
     """archspec command line interface for CPU"""
     click.echo(archspec.cpu.host())
+
+
+@main.command()
+@click.option(
+    "--cpu",
+    "only_cpu",
+    is_flag=True,
+    default=False,
+    help="Only print DAG for CPU microarchitectures",
+)
+def graph(only_cpu):
+    """Print Direct Acyclic Graph (DAG) for all known system aspects."""
+
+    def node_label(uarch):
+        """Create node label for specified Microarchitecture instance."""
+        res = uarch.name
+        if uarch.parents and uarch.vendor != uarch.parents[0].vendor:
+            res += " (" + uarch.vendor + ")"
+        return res
+
+    # for now only CPU microarchitectures are supported so this looks a bit silly,
+    # but eventually we want to print all aspects if no specific aspect was selected
+    if not only_cpu:
+        all_aspects = True
+
+    if only_cpu or all_aspects:
+        cpu_uarch_dag = graphviz.Digraph()
+
+        for uarch in archspec.cpu.TARGETS.values():
+            cpu_uarch_dag.node(uarch.name, node_label(uarch))
+            for parent in uarch.parents:
+                cpu_uarch_dag.edge(parent.name, uarch.name)
+
+        click.echo(cpu_uarch_dag.source)
```

Comment:
If we're just echoing dot source here, it's pretty easy to avoid the dependence on graphviz and spit out dot as text.  Should we?  Or is it worth having graphviz for actually rendering things later?